src/ServiceInsight/ServiceControl/DefaultServiceControl.cs

```diff
@@ -480,7 +483,17 @@ T ProcessResponse<T>(Func<IRestResponse, T> selector, IRestResponse response)
             return new List<KeyValuePair<string, string>>();
         }
 
-        static string CleanupBodyString(string bodyString) => bodyString.StartsWith(byteOrderMarkUtf8) ? bodyString.Remove(0, byteOrderMarkUtf8.Length) : bodyString;
+        static void CleanResponse(IRestResponse response)
+        {
+            if (response.RawBytes.StartsWith(byteOrderMarkUtf8))
+            {
+                var strippedBytes = response.RawBytes.Skip(byteOrderMarkUtf8.Length)
+                                            .Take(response.RawBytes.Length - byteOrderMarkUtf8.Length)
```

Comment:
The call to `Take()` is redundant. The `Skip()` will skip the BOM and then the enumerator will continue until the end of the byte array.